packages/account/src/providers/provider.ts

```diff
@@ -451,6 +451,29 @@ export default class Provider {
     }
   }
 
+  private static handleBasicAuth(url: string): [string, ProviderOptions['requestMiddleware']] {
+    const parsedUrl = new URL(url);
+
+    const username = parsedUrl.username;
+    const password = parsedUrl.password;
+    const urlNoBasicAuth = `${parsedUrl.origin}${parsedUrl.pathname}`;
+    if (!(username && password)) {
+      return [url, undefined];
+    }
+
+    const auth = `Basic ${btoa(`${username}:${password}`)}`;
+
+    return [
+      urlNoBasicAuth,
+      (request) => {
+        request.headers ??= {};
+        (request.headers as Record<string, unknown>).Authorization = auth;
+
+        return request;
+      },
+    ];
```

Comment:
Removing side-effects, keeping it simple.

```suggestion
    return { url, auth };
```